src/app/workflow/workflow.component.ts

```diff
@@ -253,6 +260,9 @@ export class WorkflowComponent extends Entry implements AfterViewInit {
       }
       this.setUpWorkflow(workflow);
     });
+    this.tokenQuery.tokens$.subscribe((tokens: Token[]) => {
+      this.zenodoAccountIsLinked = tokens.some(token => token.tokenSource === 'zenodo.org');
```

Comment:
Change `'zenodo.org'` to `Provider.ZENODO` (sorry, missed in my previous review).

Reply:
not applicable since code has been refactored according to Gary's suggestions